source/common/upstream/ring_hash_lb.h

```diff
@@ -34,25 +35,46 @@ class RingHashLoadBalancer : public ThreadAwareLoadBalancerBase,
   };
 
   struct Ring : public HashingLoadBalancer {
-    Ring(const absl::optional<envoy::api::v2::Cluster::RingHashLbConfig>& config,
-         const HostVector& hosts);
+    Ring(const HostsPerLocality& hosts_per_locality,
+         const LocalityWeightsConstSharedPtr& locality_weights,
+         const absl::optional<envoy::api::v2::Cluster::RingHashLbConfig>& config);
 
     // ThreadAwareLoadBalancerBase::HashingLoadBalancer
     HostConstSharedPtr chooseHost(uint64_t hash) const override;
 
     std::vector<RingEntry> ring_;
+
+    static const uint64_t DEFAULT_MIN_RING_SIZE = 1024;
```

Comment:
Tiny nit: the more recent Envoy style for consts is `DefaultMinRingSize`; there are still a lot of exceptions, but prefer that one.